MainWindow.xaml.cs

```diff
@@ -219,5 +220,47 @@ private void UpdateEditWindows()
                 _edit_metadata_window.Content = user_control;
             }
         }
+
+        #region Test
+        public void TestInputPack(Data.DynamicInputPack inputPack)
+        {
+            foreach (Data.HostDevice device in inputPack.HostDevices)
+            {
+                foreach (Data.HostKey key in device.HostKeys)
+                {
+                    if (!File.Exists(key.TexturePath))
+                    {
+                        key.TexturePath = GetNewImage(key.Name, key.TexturePath);
+                    }
+                }
+            }
+            foreach (Data.DynamicInputTexture texture in inputPack.Textures)
+            {
+                if (!File.Exists(texture.TexturePath))
+                {
+                    texture.TexturePath = GetNewImage(texture.TextureHash, texture.TexturePath);
+                }
+            }
+        }
+
+        private string GetNewImage(string name, string paht)
```

Comment:
May we change this to `EnsurePathExists`?  The current text `GetNewImage` makes me think that we're always getting a new image which isn't necessarily true.